packages/spanner/src/transaction.js

```diff
@@ -283,11 +314,28 @@ Transaction.prototype.commit = function(callback) {
  * @param {function} callback - The callback function.
  */
 Transaction.prototype.request = function(config, callback) {
+  var self = this;
+
   var reqOpts = extend({
     session: this.session.formattedName_
   }, config.reqOpts);
 
-  config.method(reqOpts, callback);
+  var gaxOptions = reqOpts.gaxOptions;
+  delete reqOpts.gaxOptions;
+
+  config.method(reqOpts, gaxOptions, function(err, resp) {
```

Comment:
Sorry for the late response. I was busy last week at I/O.
As @callmehiphop said, a stream failing with aborted should be treated like everything else and we should just retry the whole transaction. @stephenplusplus I am not sure what do you mean by "mixing valid and invalid responses". 
This does not seem any different from the scenario where user makes two read requests in the transaction and one of them succeed completely and the second one fails with aborted error. Even in that case if they have made some non transactional changes using the results of the previous stream, they can run into the scenario where they replay those changes when the first stream is retried. We can add the public method that allows users to do there own retry when someone asks for it or if users report issues using the automatic retry method.